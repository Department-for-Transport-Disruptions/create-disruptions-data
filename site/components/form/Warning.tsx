import { ReactElement } from "react";

interface WarningProps {
    text: string;
    symbol?: string;
    title?: string;
}

const Warning = ({ text, symbol = "!", title = "Warning" }: WarningProps): ReactElement => (
    <div className="govuk-warning-text">
        <span className="govuk-warning-text__icon" aria-hidden="true">
            {symbol}
        </span>
        <strong className="govuk-warning-text__text">
            <span className="govuk-visually-hidden">{title}</span>
            {text}
        </strong>
    </div>
);

export default Warning;
