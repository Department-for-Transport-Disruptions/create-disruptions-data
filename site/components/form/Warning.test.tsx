import { render } from "@testing-library/react";
import { describe, expect, it } from "vitest";
import Warning from "./Warning";

describe("Warning", () => {
    it("should render correctly", () => {
        const { asFragment } = render(<Warning text="Test Warning Message" />);
        expect(asFragment()).toMatchSnapshot();
    });

    it("should render correctly with optional props", () => {
        const { asFragment } = render(<Warning text="Test Warning Message" symbol="?" title="WARNING!!!" />);
        expect(asFragment()).toMatchSnapshot();
    });
});
