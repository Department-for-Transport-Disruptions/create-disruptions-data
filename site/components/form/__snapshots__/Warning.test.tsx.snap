// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Warning > should render correctly 1`] = `
<DocumentFragment>
  <div
    class="govuk-warning-text"
  >
    <span
      aria-hidden="true"
      class="govuk-warning-text__icon"
    >
      !
    </span>
    <strong
      class="govuk-warning-text__text"
    >
      <span
        class="govuk-visually-hidden"
      >
        Warning
      </span>
      Test Warning Message
    </strong>
  </div>
</DocumentFragment>
`;

exports[`Warning > should render correctly with optional props 1`] = `
<DocumentFragment>
  <div
    class="govuk-warning-text"
  >
    <span
      aria-hidden="true"
      class="govuk-warning-text__icon"
    >
      ?
    </span>
    <strong
      class="govuk-warning-text__text"
    >
      <span
        class="govuk-visually-hidden"
      >
        WARNING!!!
      </span>
      Test Warning Message
    </strong>
  </div>
</DocumentFragment>
`;
